draft-ietf-httpbis-sfbis.md

```diff
@@ -695,8 +695,10 @@ Given a string of Unicode characters as input_string, return an ASCII string sui
    1. If byte is %x25 ("%"), append "%25" to encoded_string.
    2. If byte is in the ranges %x00-1f or %x7f-ff, apply the percent-encoding defined in {{Section 2.1 of URI}} to byte and append the result to encoded_string.
    3. Otherwise, decode byte as an ASCII character and append the result to encoded_string.
-3. Let formatted_string be the result of running Serialising a String ({{ser-string}}) with encoded_string.
-4. Return the character "%" followed by formatted_string.
+3. Let output be a string containing %x25 ("%") followed by DQUOTE.
```

Comment:
If we use the definition from the URI spe, shouldn't we mandate either upper or lowercase?

Reply:
Should parsing fail if that case isn't seen?

Reply:
which case?

Reply:
whichever we specify...

Reply:
Maybe talk about different things? Parsing percent-decoded as per URI spec can fail, and we need to say something about that, no?

Reply:
I thought you were asking if we could mandate a particular casing here (in the serialisation algorithm); I was asking if that casing should be enforced by the parsing algorithm.

Reply:
Ah. Yes, it should.

Reply:
I'm a bit concerned that some producers might not be able to control the casing of their output. While they could run it through an additional step of processing, that would increase overhead for them.

Do we expect anyone to be doing something (eg security scanning) that would be expecting a particular casing, and wouldn't be able to adapt to two possible casings? My initial reaction is that I'd rather not increasing efficiency for those parties by decreasing efficiency for others...

Reply:
It's mostly a matter of consistency with the remainder of the spec. For values, there's exactly one value to serialize them (or maybe that's not the case because of Decimal???).

Reply:
We don't require sytems to error when parsing eg `0001`. Byte sequences don't require parsers to fail on padding irregularities. I appreciate that we specify error handling very carefully in this document, but we do so when there's a point to it, not merely for consistency.